client/components/domains/contact-details-form-fields/index.jsx

```diff
@@ -0,0 +1,472 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+import PropTypes from 'prop-types';
+import React, { Component, createElement } from 'react';
+import { connect } from 'react-redux';
+import {
+	noop,
+	get,
+	deburr,
+	kebabCase,
+	pick,
+	head,
+	isEqual,
+	isEmpty,
+	camelCase,
+	identity,
+} from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getCountryStates } from 'state/country-states/selectors';
+import { CountrySelect, Input, HiddenInput } from 'my-sites/domains/components/form';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormFooter from 'my-sites/domains/domain-management/components/form-footer';
+import FormButton from 'components/forms/form-button';
+import FormPhoneMediaInput from 'components/forms/form-phone-media-input';
+import { countries } from 'components/phone-input/data';
+import { forDomainRegistrations as countriesListForDomainRegistrations } from 'lib/countries-list';
+import formState from 'lib/form-state';
+import analytics from 'lib/analytics';
+import { toIcannFormat } from 'components/phone-input/phone-number';
+import NoticeErrorMessage from 'my-sites/checkout/checkout/notice-error-message';
+import GAppsFieldset from './custom-form-fieldsets/g-apps-fieldset';
+import RegionAddressFieldsets from './custom-form-fieldsets/region-address-fieldsets';
+import notices from 'notices';
+import support from 'lib/url/support';
+
+const countriesList = countriesListForDomainRegistrations();
+
+export class ContactDetailsFormFields extends Component {
+	static propTypes = {
+		eventFormName: PropTypes.string,
+		contactDetails: PropTypes.shape( {
```

Comment:
I like less repetition :) It makes sense, since they are, and must be, equal. 

We'd have to pull `this.fieldNames` out of the constructor. Making it another static var would work (and we could access it through `ContactDetailsFormFields.fieldNames`).

But now I reflect on it, `this.fieldNames` is a definitive and immutable list of form fields. It might work better as a constant. 👍

Reply:
Ah, right, I did not think about the static/instance problem, but yup, you're right - it's a constant/immutable list anyway.